api/v1alpha1/tykoasapidefinition_types.go

```diff
@@ -48,6 +52,38 @@ type TykOASReference struct {
 	ConfigmapRef ConfigMapReference `json:"configmapRef"`
 }
 
+type TykOASVersioning struct {
+	// Default contains the default version name if a request is issued without a version.
+	Default string `json:"default"`
+
+	// Enabled is a boolean flag, if set to true it will enable versioning of the API.
+	Enabled bool `json:"enabled"`
+
+	// Key contains the name of the key to check for versioning information.
+	Key string `json:"key"`
+
+	// Location contains versioning location information. It can be one of the following:
+	// header, url-param, url.
+	Location string `json:"location"`
```

Comment:
if `location` only accepts `header`, `url-param` or `url`, it is worth to add some validation here. Please see: https://book.kubebuilder.io/reference/markers/crd-validation#crd-validation